app/scripts/background.js

```diff
@@ -91,16 +92,60 @@ const PHISHING_WARNING_PAGE_TIMEOUT = ONE_SECOND_IN_MILLISECONDS;
 /**
  * In case of MV3 we attach a "onConnect" event listener as soon as the application is initialised.
  * Reason is that in case of MV3 a delay in doing this was resulting in missing first connect event after service worker is re-activated.
+ *
+ * @param remotePort
  */
-
 const initApp = async (remotePort) => {
   browser.runtime.onConnect.removeListener(initApp);
   await initialize(remotePort);
   log.info('MetaMask initialization complete.');
 };
 
+const throwErrorIfLastErrorFound = () => {
+  const err = checkForError();
+  if (err) {
+    throw err;
+  }
+};
+
+/**
+ * Sends a message to the dapp(s) content script to signal it can connect to MetaMask background as
+ * the backend is not active. It is required to re-connect dapps after service worker re-activates.
+ * For non-dapp pages, the message will be sent and ignored.
+ */
+const sendReadyMessageToTabs = async () => {
+  try {
+    const tabs = await browser.tabs.query({});
+    throwErrorIfLastErrorFound();
```

Comment:
We can probably create a `browser...` utility that iinternally takes care to `throwErrorIfLastErrorFound`.

Reply:
Not blocker for this PR.

Reply:
updated. see https://github.com/MetaMask/metamask-extension/pull/16075#discussion_r1013279850